In [1]:
import spektral

In [3]:
spektral.datasets.graphsage.PPI() # download PPI dataset 

100%|██████████████████████████████████████| 25.8M/25.8M [00:12<00:00, 2.08MB/s]


Processing dataset.


PPI(n_graphs=1)

In [7]:
from spektral.datasets.graphsage import PPI
dataset = PPI()

print(dataset) # number of graphs 

PPI(n_graphs=1)


In [11]:
graph = dataset[0]
print(f"Total nodes: {graph.x.shape[0]}")
print(f"Total node features: {graph.x.shape[1]}")
print(f"Number of labels per node: {graph.y.shape[1]}")
print(f"Adjacency matrix shape: {graph.a.shape}")

Total nodes: 56944
Total node features: 50
Number of labels per node: 121
Adjacency matrix shape: (56944, 56944)


Since the PPI data comes with a train-test-validation split on its own - Explore the distribution 

In [14]:
mask_tr = dataset.mask_tr
mask_va = dataset.mask_va 
mask_te = dataset.mask_te  

x_train = graph.x[mask_tr]
a_train = graph.a
y_train = graph.y[mask_tr]

x_val = graph.x[mask_va]
y_val = graph.y[mask_va]

x_test = graph.x[mask_te]
y_test = graph.y[mask_te]


print(f"Train set features: {x_train.shape}")
print(f"Val set features: {x_val.shape}")
print(f"Test set features: {x_test.shape}")


Train set features: (44906, 50)
Val set features: (6514, 50)
Test set features: (5524, 50)


## PPI Dataset 

The PPI dataset contains a **single large graph** in this case - Split into a Train-Test-Validation Set 

### Training Set
The training set have 44,906 nodes. Each node/protein is represented by a 50-dimensional feature vector. These features are gene sets, motif gene sets, etc
Each of the 44,906 nodes have 121 possible labels.


Similarly for the Validation and Test set 

In [15]:
import numpy as np
import matplotlib.pyplot as plt

feature_means = np.mean(graph.x, axis=0)
feature_stds = np.std(graph.x, axis=0)

plt.figure(figsize=(10, 5))
plt.subplot(1, 2, 1)
plt.bar(range(graph.x.shape[1]), feature_means)
plt.title('Feature Means')
plt.subplot(1, 2, 2)
plt.bar(range(graph.x.shape[1]), feature_stds)
plt.title('Feature Standard Deviations')
plt.show()


ModuleNotFoundError: No module named 'matplotlib'